checkpoint2/Program.cs

```diff
@@ -0,0 +1,281 @@
+        }
+
+        //allows a checker to be removed
+        public void RemoveChecker (int destRow, int destColumn) {
+            Checker cx2 = SelectCheckerDestination (destRow, destColumn);
+            Checkers.Remove (cx2);
+        }
+        //king a checker takes in a checker object and a destRow to determine if the Checker is to be Kinged on that move
+        public void KingChecker (Checker checker, int destRow) {
+            if ((checker.Color == "black" && destRow == 7) || (checker.Color == "white" && destRow == 0)) {
+                checker.King = true;
+                if (checker.Color == "black") {
+                    int kingId = int.Parse ("24C7", System.Globalization.NumberStyles.HexNumber);
+                    string king = char.ConvertFromUtf32 (kingId);
+                    checker.Symbol = king;
+                } else {
+                    int kingId = int.Parse ("24D1", System.Globalization.NumberStyles.HexNumber);
+                    string king = char.ConvertFromUtf32 (kingId);
+                    checker.Symbol = king;
+                }
+                PlaceCheckers ();
+            }
+        }
+
+        public void MoveCheckers () {
+
+            Console.WriteLine ("(Choose a number between 0-7)");
+            Console.WriteLine ("Pick up Checker on Row: ");
+            int sourceRow = Int32.Parse (Console.ReadLine ().Trim ());
+            Console.WriteLine ("Pick up Checker on Column: ");
+            int sourceColumn = Int32.Parse (Console.ReadLine ().Trim ());
+            Checker cx = SelectCheckerSource (sourceRow, sourceColumn);
+            Console.WriteLine ("Place Checker on Row: ");
+            int destRow = Int32.Parse (Console.ReadLine ().Trim ());
+            Console.WriteLine ("Place Checker on Column: ");
+            int destColumn = Int32.Parse (Console.ReadLine ().Trim ());
+            CheckerPreCheck (sourceRow, sourceColumn, destRow, destColumn, check);
+            cx.Position = new int[] { destRow, destColumn };
+            // check for a king 
+            KingChecker (cx, destRow);
+            //creates the board again
+            CreateBoard ();
+            //places checkers on the newly created board
+            PlaceCheckers ();
+            return;
+        }
+
+    }
+
+    class Game {
+        public Game () {
+            // determines turns
+            bool turn = true;
+
+            //initialize the game by creating a board object and apply board methods
+            Board board = new Board ();
+            board.CreateBoard ();
+            board.GenerateCheckers ();
+            board.PlaceCheckers ();
+            Console.WriteLine ("To play you will select rows and columns\nto move pieces throughout the field.");
+            do {
+                //draw the board to play. on loop checkers will remember they have new positions.
+                //try again for try catch exceptions.
+                TryAgain : board.DrawBoard ();
```

Comment:
GOTO statements as are no no. I would rework this into a while loop that loop while input validation fails.

We have a nice high level language (C#) that helps us avoid errors, GOTO statements are generally not good.